### PR TITLE
feat(swr): support namespace permissions in sdk

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
support namespace permissions in sdk
```

### CODE REVIEW

File: openstack/swr/v2/namespaces/requests.go [hunk 1]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
 	})
 	return
 }
+
+// CreateAccessOptsBuilder allows extensions to add additional parameters to the
+// Create request.
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:58]: use one line to explain

File: openstack/swr/v2/namespaces/requests.go [hunk 2]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
+	}
+	reqOpt := &golangsdk.RequestOpts{OkCodes: []int{201}}
+	_, r.Err = c.Post(accessURL(c, nameSpace), b["Users"], &r.Body, reqOpt)
+	return
+}
+
+// Get retrieves the access of a namespace.
+func GetAccess(c *golangsdk.ServiceClient, nameSpace string) (r GetAccessResult) {
+	_, r.Err = c.Get(accessURL(c, nameSpace), &r.Body, &golangsdk.RequestOpts{
+		OkCodes:     []int{200},
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:98]: 200 is already in the default response codes for GET method.

File: openstack/swr/v2/namespaces/requests.go [hunk 3]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
+}
+
+// CreateAccess accepts a CreateAccessOpts struct and uses the values to create access of a namespace.
+func CreateAccess(c *golangsdk.ServiceClient, opts CreateAccessOptsBuilder, nameSpace string) (r CreateAccessResult) {
+	b, err := opts.ToAccessCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	reqOpt := &golangsdk.RequestOpts{OkCodes: []int{201}}
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:90]: 201 is already in the default response codes for POST method.

File: openstack/swr/v2/namespaces/urls.go [hunk 4]
```diff
@@ -9,3 +9,7 @@ func rootURL(client *golangsdk.ServiceClient) string {
 func resourceURL(client *golangsdk.ServiceClient, name string) string {
 	return client.ServiceURL("manage", "namespaces", name)
 }
+
+func accessURL(client *golangsdk.ServiceClient, nameSpace string) string {
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/urls.go:13]: ditto

File: openstack/swr/v2/namespaces/requests.go [hunk 5]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
+	Users []User
+}
+
+// Access information of a user
+type User struct {
+	// ID of the user
+	UserID string `json:"user_id" required:"true"`
+	// Name of the user
+	UserName string `json:"user_name" required:"true"`
+	// Permission of the user
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:74]: Missing permission description:
7: manage
3: write
1: read

File: openstack/swr/v2/namespaces/requests.go [hunk 6]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
+	reqOpt := &golangsdk.RequestOpts{OkCodes: []int{201}}
+	_, r.Err = c.Post(accessURL(c, nameSpace), b["Users"], &r.Body, reqOpt)
+	return
+}
+
+// Get retrieves the access of a namespace.
+func GetAccess(c *golangsdk.ServiceClient, nameSpace string) (r GetAccessResult) {
+	_, r.Err = c.Get(accessURL(c, nameSpace), &r.Body, &golangsdk.RequestOpts{
+		OkCodes:     []int{200},
+		MoreHeaders: RequestOpts.MoreHeaders, JSONBody: nil,
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:99]: Nil JSONBody is unnecessary to desplay

File: openstack/swr/v2/namespaces/requests.go [hunk 7]
```diff
@@ -53,3 +53,57 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
...
+	})
+	return
+}
+
+// Delete will permanently delete the access of a namespace.
+func DeleteAccess(c *golangsdk.ServiceClient, userIDs []string, nameSpace string) (r DeleteAccessResult) {
```
Comment (Lance52259) [inline openstack/swr/v2/namespaces/requests.go:105]: Use `namespace` because it is a word.

**Review by Lance52259 — CHANGES_REQUESTED** [review]

File: openstack/swr/v2/namespaces/requests.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

File: openstack/swr/v2/namespaces/requests.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

File: openstack/swr/v2/namespaces/requests.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

File: openstack/swr/v2/namespaces/urls.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/urls.go]: fixed

File: openstack/swr/v2/namespaces/requests.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

File: openstack/swr/v2/namespaces/requests.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

File: openstack/swr/v2/namespaces/requests.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply openstack/swr/v2/namespaces/requests.go]: fixed

**Pull request merged**